components/eam/src/physics/cam/ndrop.F90

```diff
@@ -503,208 +479,200 @@ subroutine dropmixnuc( &
 
-   if (prog_modal_aero) then
-      ! aerosol tendencies
-      call physics_ptend_init(ptend, state%psetcols, 'ndrop_aero', lq=lq)
-   else
-      ! no aerosol tendencies
-      call physics_ptend_init(ptend, state%psetcols, 'ndrop')
-   end if
+   ! Init pointers to mode number and species mass mixing ratios for
+   ! cloud borne phases.
+
+   do imode = 1, ntot_amode
+      mm = mam_idx(imode, 0)
+      num_idx = numptrcw_amode(imode)
+      qqcw(mm)%fld => qqcw_get_field(pbuf,num_idx,lchnk,.true.)
+      do lspec = 1, nspec_amode(imode)
+         mm = mam_idx(imode, lspec)
+         spc_idx = lmassptrcw_amode(lspec,imode)
+         qqcw(mm)%fld => qqcw_get_field(pbuf,spc_idx,lchnk,.true.)
+      enddo
+   enddo
+
+   arg = 1.0_r8
+   if (abs(0.8427_r8 - erf(arg))/0.8427_r8 > 0.001_r8) then
+      write(iulog,*) 'erf(1.0) = ',ERF(arg)
+      call endrun('dropmixnuc: Error function error')
+   endif
+   arg = 0.0_r8
+   if (erf(arg) /= 0.0_r8) then
+      write(iulog,*) 'erf(0.0) = ',erf(arg)
+      write(iulog,*) 'dropmixnuc: Error function error'
+      call endrun('dropmixnuc: Error function error')
+   endif
+
+   dtinv = 1._r8/dtmicro
 
    !initialize variables to zero
    ndropmix(:,:) = 0._r8
    nsource(:,:) = 0._r8
+   wtke(:,:)    = 0._r8
+   factnum(:,:,:) = 0._r8
 
-   ! overall_main_i_loop
-   do i = 1, ncol
+   ! overall_main_icol_loop
+   do icol = 1, ncol
 
-      do k = top_lev, pver-1
-         zs(k) = 1._r8/(zm(i,k) - zm(i,k+1))
-      end do
+      do kk = top_lev, pver-1
+         zs(kk) = 1._r8/(zm(icol,kk) - zm(icol,kk+1))
+      enddo
       zs(pver) = zs(pver-1)
 
       ! load number nucleated into qcld on cloud boundaries
 
-      do k = top_lev, pver
-
-         qcld(k)  = ncldwtr(i,k)
-         qncld(k) = 0._r8
-         srcn(k)  = 0._r8
-         cs(i,k)  = pmid(i,k)/(rair*temp(i,k))        ! air density (kg/m3)
-         dz(i,k)  = 1._r8/(cs(i,k)*gravit*rpdel(i,k)) ! layer thickness in m
-
-         do m = 1, ntot_amode
-            nact(k,m) = 0._r8
-            mact(k,m) = 0._r8
-         end do
-
-         zn(k) = gravit*rpdel(i,k)
-
-         if (k < pver) then
-            ekd(k)   = kvh(i,k+1)
-            ekd(k)   = max(ekd(k), zkmin)
-            ekd(k)   = min(ekd(k), zkmax)
-            csbot(k) = 2.0_r8*pint(i,k+1)/(rair*(temp(i,k) + temp(i,k+1)))
-            csbot_cscen(k) = csbot(k)/cs(i,k)
+      do kk = top_lev, pver
+
+         cs(icol,kk)  = pmid(icol,kk)/(rair*temp(icol,kk))        ! air density (kg/m3)
+         dz(icol,kk)  = 1._r8/(cs(icol,kk)*gravit*rpdel(icol,kk)) ! layer thickness in m
+         zn(kk) = gravit*rpdel(icol,kk)
+         qcld(kk)  = ncldwtr(icol,kk)
+         qncld(kk) = 0._r8
+         srcn(kk)  = 0._r8
+         nact(kk,1:ntot_amode) = 0._r8
+         mact(kk,1:ntot_amode) = 0._r8
+
+         if (kk < pver) then
+            ekd(kk)   = kvh(icol,kk+1)
+            ekd(kk)   = max(ekd(kk), zkmin)
+            ekd(kk)   = min(ekd(kk), zkmax)
```

Comment:
This may use min_max_bound subroutine

Reply:
Done.

Reply:
It looks that the recent commit does not include this change. Can you check it? @brigaudet

Reply:
Sorry, it should be there now.